test/e2e/node_ip_migration.go

```diff
@@ -0,0 +1,584 @@
+
+		By("Removing the egress assignable label from all nodes")
+		// true here means only select worker nodes
+		workerNodes, err := ocGetNodes(true)
+		Expect(err).NotTo(HaveOccurred())
+		for nodeName := range workerNodes {
+			_, _ = framework.RunKubectl("default", "label", "node", nodeName, "k8s.ovn.org/egress-assignable-")
+		}
+	})
+
+	Context("when the node IP address is updated", func() {
+		It("makes sure that the cluster is still operational", func() {
+			By("Retrieving the test pods")
+			podWorkerNode, err := fr.PodClient().Get(context.TODO(), podWorkerNodeName, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			podSecondaryWorkerNode, err := fr.PodClient().Get(context.TODO(), podSecondaryWorkerNodeName, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			By(fmt.Sprintf("Poking from %s on node %s to %s on node %s for a maximum of %d seconds",
+				podWorkerNode.Name, podWorkerNode.Spec.NodeName,
+				podSecondaryWorkerNode.Name, podSecondaryWorkerNode.Spec.NodeName,
+				pollingTimeout))
+			Eventually(func() bool {
+				By("Poking the pod")
+				err := pokePod(fr, podWorkerNodeName, podSecondaryWorkerNode.Status.PodIP)
+				return err == nil
+			}, pollingTimeout, pollingInterval).Should(BeTrue())
+
+			By("Verifying egress IP still works")
+			expectedAnswer := "^" + egressIp + ":.*"
+			if utilnet.IsIPv6String(egressIp) {
+				expectedAnswer = "^\\[" + egressIp + "\\]:.*"
+			}
+			Eventually(func() bool {
+				By("Checking the egress IP")
+				res, err := targetExternalContainerConnectToEndpoint(
+					externalContainerName,
+					externalContainerIp,
+					externalContainerPort,
+					externalContainerEndpoint,
+					podWorkerNode.Name,
+					fr.Namespace.Name,
+					true,
+					expectedAnswer,
+				)
+				if err != nil {
+					framework.Logf("Current verification failed with %s", err)
+					return false
+				}
+				return res
+			}, pollingTimeout, pollingInterval).Should(BeTrue())
+		})
+
+	})
+})
+
+// ocGetNodes returns a map[string]string that maps nodeNames to IP addresses.
+// When workersOnly is true, return only worker nodes (nodes with label node-role.kubernetes.io/master!="").
+func ocGetNodes(workersOnly bool) (map[string]string, error) {
+	nodeMap := make(map[string]string)
+	cmd := []string{
+		"get", "nodes", "-o=jsonpath='{range.items[*]}{.metadata.name} {\"\\t\"} {.status.addresses[?(@.type==\"InternalIP\")].address}{\"\\n\"}{end}'"}
+	if workersOnly {
+		cmd = append(cmd, "-l", "node-role.kubernetes.io/master!=")
+	}
+	output, err := framework.RunKubectl("", cmd...)
+	if err != nil {
+		return nil, err
+	}
+	output = strings.Trim(output, "'")
+	scanner := bufio.NewScanner(strings.NewReader(output))
+	for scanner.Scan() {
+		s := strings.Split(scanner.Text(), "\t")
+		if len(s) != 2 {
+			continue
+		}
+		nodeMap[strings.TrimSpace(s[0])] = strings.TrimSpace(s[1])
+	}
+	return nodeMap, nil
+}
+
+// findNextWorkerNodeIP will set the worker's IP to the next higher available IP address.
```

Comment:
I think potentially in the long run this would clash if we brought up an additional docker container on the same network because docker would just assign the next ip to the next container. It might be safer to grab the last usable IP in the subnet, then decrement instead.

Reply:
Now I realized you use last ips for egress IPs. Maybe we can just reserve a few of the last ips in test setup to use.